Dependencies/BabylonNativeUtils/Source/NetworkUtils.mm

```diff
@@ -0,0 +1,54 @@
+#include "NetworkUtils.h"
+#import <Foundation/Foundation.h>
+
+namespace Babylon
+{
+    std::string GetAbsoluteUrl(const std::string& url, const std::string& rootUrl)
+    {
+        NSString *urlStr = [NSString stringWithCString:url.c_str() encoding:[NSString defaultCStringEncoding]];
+        NSString *rootUrlStr = [NSString stringWithCString:rootUrl.c_str() encoding:[NSString defaultCStringEncoding]];
+        NSString *completeURL = [NSString stringWithFormat:@"%@/%@",rootUrlStr,urlStr];
+        NSURL *baseUrl = [NSURL URLWithString:completeURL];
+        NSError *error;
+        BOOL reachable = [baseUrl checkResourceIsReachableAndReturnError:&error];
+        if (reachable)
+        {
+            return std::string([completeURL UTF8String]);
+        }
+        return url;
+    }
+
+    template<typename DataT>
+    arcana::task<DataT, std::exception_ptr> LoadUrlAsync(std::string url)
+    {
+        __block arcana::task_completion_source<DataT, std::exception_ptr> taskCompletionSource{};
+        NSString *urlStr = [NSString stringWithCString:url.c_str() encoding:[NSString defaultCStringEncoding]];
+        NSURLRequest *request = [NSURLRequest requestWithURL:[NSURL URLWithString:urlStr]];
+        NSURLSession *session = [NSURLSession sharedSession];
+        NSURLSessionDataTask *task = [session dataTaskWithRequest:request completionHandler:^(NSData * _Nullable data, NSURLResponse * _Nullable response, NSError * _Nullable error) {
+            if (!error) {
+                DataT dataT{};
+                dataT.resize(data.length);
+                [data getBytes:dataT.data()
+                        length:data.length];
```

Comment:
Does this handle text conversion correctly?

Reply:
in XMLHttpRequest, the conversion UTF8-BOM to UTF8 is handled correctly thanks to the NSString conversion. In NetworkUtils, there is no conversion. It read raw bytes from the network.

Reply:
Isn't that going to cause issues?

Reply:
Yes. If you try to load and evaluate an UTF8-BOM .js you'll get a parsing error. On all the platforms we support.

Reply:
Can we not do the same thing as what XMLHttpRequest is doing?

Reply:
It would be easier if we can share code between webrequest/xmlhttprequest and NetworkUtils. Life is more difficult with Android. As we have to call Java from jni, the network code has to be part of the testapp/integration aar. So this will revisited.